test/coverage/datamodel/UMessageValidatorTest.cpp

```diff
@@ -770,4 +770,94 @@ TEST_F(TestUMessageValidator, ValidNotification) {
 	}
 }
 
+TEST_F(TestUMessageValidator, IsValid) {
+	{
+		// valid request
+		auto source = source_;
+		auto sink = sink_;
+		source.set_resource_id(0);
+
+		auto attributes = fakeRequest(source, sink);
+		auto umessage = build(attributes);
+		auto [valid, reason] = isValid(umessage);
+		EXPECT_TRUE(valid);
+		EXPECT_FALSE(reason.has_value());
+
+		// Make the request invalid by making one change
+		attributes.set_ttl(0);
+		umessage = build(attributes);
+
+		auto [valid2, reason2] = isValid(umessage);
+		EXPECT_FALSE(valid2);
+		EXPECT_TRUE(reason2.has_value());
+		EXPECT_EQ(reason2, Reason::INVALID_TTL);
+	}
+
+	{
+		// valid response
+		auto source = source_;
+		auto sink = sink_;
+		source.set_resource_id(0);
+
+		auto attributes = fakeResponse(source, sink);
+		auto umessage = build(attributes);
+		auto [valid, reason] = isValidRpcResponse(umessage);
+		EXPECT_TRUE(valid);
+		EXPECT_FALSE(reason.has_value());
+
+		// Make the request invalid by making one change
+		attributes = fakeResponse(sink, source);
+		umessage = build(attributes);
+
+		auto [valid2, reason2] = isValid(umessage);
+		EXPECT_FALSE(valid2);
+		EXPECT_TRUE(reason2.has_value());
+		EXPECT_EQ(reason2, Reason::BAD_SOURCE_URI);
+	}
+
+	{
+		// valid publish
+		auto source = source_;
+		source.set_resource_id(0x8000);
+
+		auto attributes = fakePublish(source);
+		auto umessage = build(attributes);
+		auto [valid, reason] = isValidPublish(umessage);
```

Comment:
Function being called: IsValidPublish -> isValid

Honestly both wouldn't hurt, but that is a bit of over-testing in my opinion.

Reply:
Good catch. Fixed